Lib/gftools/builder/__init__.py

```diff
@@ -0,0 +1,341 @@
+            for file_name in file_names:
+                shutil.move(
+                    os.path.join(args["output_dir"], file_name), original_output_dir
+                )
+            tmpdir.cleanup()
+            args["output_dir"] = original_output_dir
+            return [os.path.join(original_output_dir, x) for x in file_names]
+
+    def rename_variable(self, fontfile):
+        if "axisOrder" not in self.config:
+            font = TTFont(fontfile)
+            self.config["axisOrder"] = [ax.axisTag for ax in font["fvar"].axes]
+        axes = ",".join(self.config["axisOrder"])
+        newname = fontfile.replace("-VF.ttf", "[%s].ttf" % axes)
+        os.rename(fontfile, newname)
+        return newname
+
+    def gen_stat(self, varfonts):
+        filenames = varfonts.values()
+        if len(filenames) > 1 and "ital" not in self.config["axisOrder"]:
+            # *Are* any italic? Presumably, but test
+            if any([ "italic" in x.lower() for x in filenames ]):
+                self.config["axisOrder"].append("ital")
+
+        if "stylespaceFile" in self.config and self.config["stylespaceFile"]:
+            self.gen_stat_stylespace(self.config["stylespaceFile"], filenames)
+        elif "stat" in self.config:
+            self.gen_stat_config(self.config["stat"], varfonts)
+        else:
+            ttFonts = [TTFont(x) for x in filenames]
+            gen_stat_tables(ttFonts, self.config["axisOrder"])
+            for filename, ttFont in zip(filenames, ttFonts):
+                ttFont.save(filename)
+
+    def gen_stat_config(self, stat, varfonts):
+        # Check we have no italic
+        if isinstance(stat, list):
+            if "ital" in self.config["axisOrder"]:
+                for ax in stat:
+                    if ax["name"] == "ital":
+                        raise ValueError("ital axis should not appear in stat config")
+            ital_stat_for_roman = {
+                "name": "Italic", "tag": "ital",
+                "values": [dict(value=0, name="Roman", flags=0x2, linkedValue=1)]
+            }
+            ital_stat_for_italic = {
+                "name": "Italic", "tag": "ital",
+                "values": [dict(value=1, name="Italic")]
+            }
+
+            stat.append({}) # We will switch this entry between Roman and Italic
+
+        for source, f in varfonts.items():
+            ttFont = TTFont(f)
+            if isinstance(stat, dict):
+                if source not in stat:
+                    raise ValueError("Source %s not found in stat dictionary" % source)
+                this_stat = stat[source]
+            else:
+                if "italic" in f.lower():
+                    stat[-1] = ital_stat_for_italic
+                else:
+                    stat[-1] = ital_stat_for_roman
+                this_stat = stat
+            buildStatTable(ttFont, this_stat)
+            ttFont.save(f)
+
+    def gen_stat_stylespace(self, stylespaceFile, filenames):
```

Comment:
Perhaps we should log a warning that we prefer the other methods over this one, since they provide more functionality?